packages/dcos-integration-test/extra/test_tweeter_demo.py

```diff
@@ -0,0 +1,86 @@
+import os
+import time
+
+import requests
+
+from test_packaging import (
+    _skipif_insufficient_resources
+)
+
+DEMO_RESOURCE_REQUIREMENTS = {
+    'number_of_nodes': 5,
+    'node': {
+        'disk': 0,
+        'mem': 0,
+        'cpus': 1
+    }
+}
+
+
+def _get_tweeter_app_definition():
+    response = requests.get('https://raw.githubusercontent.com/mesosphere/tweeter/master/tweeter.json')
+    return response.json()
+
+
+def _get_post_tweets_app_definition():
+    response = requests.get('https://raw.githubusercontent.com/mesosphere/tweeter/master/post-tweets.json')
+    return response.json()
+
+
+def _get_tweet_count(address):
+    """Count the number of tweets on the tweeter app's webpage
+
+    Args:
+        address: str URL including http://
+    """
+    response = requests.get(address)
+    return response.text.count('tweet-content')
+
+
+def test_tweeter_demo_oss(dcos_api_session):
+    """Step through setup and run the Tweeter demo application. See https://github.com/mesosphere/tweeter
+    Should be equivalent to the cli script. It passes if service and app installations are successful and the
+    post_tweets application successfully posts tweets to the tweeter webapp.
+
+    Service versions are read through environment variables set through the Tweeter Demo jov on TeamCity. These default
+    to None, so cosmos/packaging will install the most recent version of the service compatible with the version of
+    DC/OS being used.
+
+    Note that the Zeppelin notebook is not run, nor is it set up with cores_max parameter to 8 as noted in the docs for
+    manually running this.
+    """
+    _skipif_insufficient_resources(dcos_api_session, DEMO_RESOURCE_REQUIREMENTS)
+
+    dependencies = {
+        'cassandra': os.environ.get('CASSANDRA_VERSION'),
+        'kafka': os.environ.get('KAFKA_VERSION'),
+        'zeppelin': os.environ.get('ZEPPELIN_VERSION'),
+        'marathon-lb': os.environ.get('MARATHON_LB_VERSION')
+    }
+
+    for service, version in dependencies.items():
+        print("Installing {0} {1}".format(service, version or "(most recent version)"))
+        dcos_api_session.cosmos.install_package(service, version)
+
+    print("Waiting for requirements to deploy")
+    dcos_api_session.marathon.wait_for_deployments_complete()
+
+    tweeter_app_definition = _get_tweeter_app_definition()
+    print("Deploying tweeter")
+    print(str(tweeter_app_definition))
+    dcos_api_session.marathon.deploy_app(tweeter_app_definition)
+
+    post_tweets_definition = _get_post_tweets_app_definition()
+    print("Deploying post_tweets")
+    print(post_tweets_definition)
+    # post-tweets does not define a health check
+    dcos_api_session.marathon.deploy_app(post_tweets_definition, check_health=False)
+    # wait for some of the tweets to be posted
+    time.sleep(60)
```

Comment:
This sleep may be conservative enough to be very safe, but I think I would still argue for a simple retry loop that doesn't involve guessing might be preferable... This code through the end is functionally identical:
```
@retrying.retry(
    wait_fixed=10000,
    stop_max_delay=60000)
def _wait_for_tweet_count(url):
    assert _get_tweet_count_(url) == 25, "Did not find expected 25 tweets"
```
`_get_tweet_count` is a single purpose function, so you could just remove it completely for:
```
@retrying.retry(
    wait_fixed=10000,
    stop_max_delay=60000)
def _wait_for_tweets(url):
    assert requests.get(address).text.count('tweet-content') == 25, "Did not find expected 25 tweets"
```